### ISSUE TITLE
Crash report - startIndex darf nicht länger als die Länge der Zeichenfolge sein. Parametername: startIndex (problem with substring call)

### ISSUE THREAD

**Issue by JuliusSweetland** [JuliusSweetland/OptiKey#88]
````
<?xml version="1.0"?>
<SerializableException xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <ExtendedInformation>
    <ParamName>startIndex</ParamName>
    <HResult>-2146233086</HResult>
  </ExtendedInformation>
  <Message>startIndex darf nicht länger als die Länge der Zeichenfolge sein.
Parametername: startIndex</Message>
  <Source>mscorlib</Source>
  <StackTrace>   bei System.String.Substring(Int32 startIndex, Int32 length)
   bei JuliusSweetland.OptiKey.Services.KeyboardOutputService.SwapText(String textToSwapOut, String textToSwapIn) in c:\Users\Julius\Documents\GitHub\OptiKey\src\Services\KeyboardOutputService.cs:Zeile 725.
   bei JuliusSweetland.OptiKey.Services.KeyboardOutputService.SwapLastTextChangeForSuggestion(Int32 index) in c:\Users\Julius\Documents\GitHub\OptiKey\src\Services\KeyboardOutputService.cs:Zeile 678.
   bei JuliusSweetland.OptiKey.Services.KeyboardOutputService.ProcessFunctionKey(FunctionKeys functionKey) in c:\Users\Julius\Documents\GitHub\OptiKey\src\Services\KeyboardOutputService.cs:Zeile 187.
   bei JuliusSweetland.OptiKey.UI.ViewModels.MainViewModel.HandleFunctionKeySelectionResult(KeyValue singleKeyValue) in c:\Users\Julius\Documents\GitHub\OptiKey\src\UI\ViewModels\MainViewModel.ServiceEventHandlers.cs:Zeile 0.
   bei JuliusSweetland.OptiKey.UI.ViewModels.MainViewModel.KeySelectionResult(Nullable`1 singleKeyValue, List`1 multiKeySelection) in c:\Users\Julius\Documents\GitHub\OptiKey\src\UI\ViewModels\MainViewModel.ServiceEventHandlers.cs:Zeile 138.
   bei JuliusSweetland.OptiKey.UI.ViewModels.MainViewModel.&lt;AttachServiceEventHandlers&gt;b__6(Object o, Tuple`4 tuple) in c:\Users\Julius\Documents\GitHub\OptiKey\src\UI\ViewModels\MainViewModel.ServiceEventHandlers.cs:Zeile 112.
   bei JuliusSweetland.OptiKey.Services.InputService.PublishSelectionResult(Tuple`4 selectionResult) in c:\Users\Julius\Documents\GitHub\OptiKey\src\Services\InputService.cs:Zeile 404.
   bei JuliusSweetland.OptiKey.Services.InputService.ProcessSelectionTrigger(TriggerSignal triggerSignal) in c:\Users\Julius\Documents\GitHub\OptiKey\src\Services\InputService.subscriptions.cs:Zeile 185.
   bei System.Reactive.AnonymousSafeObserver`1.OnNext(T value)
   bei System.Reactive.Concurrency.ObserveOn`1.ObserveOnSink.OnNextPosted(Object value)
   bei System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   bei MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)</StackTrace>
  <TargetSite>System.String Substring(Int32, Int32) @ System.String</TargetSite>
  <Type>System.ArgumentOutOfRangeException</Type>
</SerializableException>
````

**Comment by evanschulte** [comment]
Any notes on how to reproduce this?  I can't imagine how the string startIndex could be less than 0 if the length of the string isn't null.

**Comment by JuliusSweetland** [comment]
@evanschulte - Not at the moment. I've only seen this where the user's local language is German. I wonder if there are german characters that could possibly be counted as more than 1 character? Sounds odd, but I half remember from school that the 'ß' character is the equivalent of a double 's'. This might just be for pronunciation, but there may be something weird like that going on.

**Comment by steinarxff** [comment]
If its German only it might be some unicode unicorns.

See: 
http://csharpindepth.com/Articles/General/Strings.aspx - scroll down and read "Encoding" and "Culture and internationalization oddities"

consider this aswell

http://stackoverflow.com/a/7471047

Good luck, and I don't think you can reproduce without a german system.

**Comment by JuliusSweetland** [comment]
Thank you. German localisation will be coming soon so I'll beef up the
logging around this and we can hopefully catch it quickly.

**Comment by JuliusSweetland** [comment]
Fixed in release v2.2.0
